### PR TITLE
BUGZ-696 Drop non DS UDP packets when silent domain checkins exceeds 3 seconds

### PR DESCRIPTION
**Pull request by roxanneskelly**
https://highfidelity.atlassian.net/browse/BUGZ-696

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19988/HighFidelity-Beta-PR15771-3d2b1db.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19988/HighFidelity-Beta-Interface-PR15771-3d2b1db.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19988/HighFidelity-Beta-PR15771-3d2b1db.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19988/HighFidelity-Quest-Beta-PR15771-3d2b1db.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19989/HighFidelity-Beta-PR15771-5b6911e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19989/HighFidelity-Quest-Beta-PR15771-5b6911e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19989/HighFidelity-Beta-PR15771-5b6911e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19989/HighFidelity-Beta-Interface-PR15771-5b6911e.exe

File: libraries/networking/src/LimitedNodeList.cpp [hunk 1]
```diff
@@ -409,6 +409,13 @@ qint64 LimitedNodeList::sendUnreliablePacket(const NLPacket& packet, const HifiS
     Q_ASSERT_X(!packet.isReliable(), "LimitedNodeList::sendUnreliablePacket",
                "Trying to send a reliable packet unreliably.");
 
+    if(_dropOutgoingNodeTraffic) {
+        auto destinationNode = findNodeWithAddr(sockAddr);
+        if (!destinationNode.isNull() && (destinationNode->getType() != NodeType::DomainServer)) {
+            return ERROR_SENDING_PACKET_BYTES;
```
Comment (SimonWalton-HiFi) [inline libraries/networking/src/LimitedNodeList.cpp]: I don't think the DS is ever returned as a Node, so just the isNull check should be enough.

File: libraries/networking/src/LimitedNodeList.cpp [hunk 2]
```diff
@@ -409,6 +409,13 @@ qint64 LimitedNodeList::sendUnreliablePacket(const NLPacket& packet, const HifiS
...
+        }
+    }
+
     fillPacketHeader(packet, hmacAuth);
```
Comment (SimonWalton-HiFi) [inline libraries/networking/src/LimitedNodeList.cpp]: There's a couple of `sendPacketList` functions also - I don't know if they are used much.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19990/HighFidelity-Beta-PR15771-2be828a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19990/HighFidelity-Quest-Beta-PR15771-2be828a.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19990/HighFidelity-Beta-PR15771-2be828a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19990/HighFidelity-Beta-Interface-PR15771-2be828a.dmg

File: domain-server/src/DomainServer.cpp [hunk 3]
```diff
@@ -1735,7 +1735,12 @@ void DomainServer::nodePingMonitor() {
     nodeList->eachNode([now](const SharedNodePointer& node) {
         quint64 lastHeard = now - node->getLastHeardMicrostamp();
         if (lastHeard > 2 * USECS_PER_SECOND) {
-            qCDebug(domain_server) << "Haven't heard from " << node->getPublicSocket() << " in " << lastHeard / USECS_PER_MSEC << " msec";
+            QString username;
+            DomainServerNodeData* nodeData = static_cast<DomainServerNodeData*>(node->getLinkedData());
+            if(nodeData) {
```
Comment (howard-stearns) [inline domain-server/src/DomainServer.cpp]: coding standard. whitespace between if and paren.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19990/HighFidelity-Beta-PR15771-2be828a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19990/HighFidelity-Beta-Interface-PR15771-2be828a.exe

File: libraries/networking/src/DomainHandler.cpp [hunk 4]
```diff
@@ -566,9 +566,14 @@ bool DomainHandler::checkInPacketTimeout() {
         qCDebug(networking_ice) << "Silent domain checkins:" << _checkInPacketsSinceLastReply;
     }
 
-    if (_checkInPacketsSinceLastReply > MAX_SILENT_DOMAIN_SERVER_CHECK_INS) {
+    auto nodeList = DependencyManager::get<NodeList>();
 
-        auto nodeList = DependencyManager::get<NodeList>();
+    if(_checkInPacketsSinceLastReply > 2) {
```
Comment (howard-stearns) [inline libraries/networking/src/DomainHandler.cpp]: coding standard:
- space between if and paren
- no magic numbers. 2 should be a constant

File: libraries/networking/src/DomainHandler.cpp [hunk 5]
```diff
@@ -566,9 +566,14 @@ bool DomainHandler::checkInPacketTimeout() {
...
+        qCDebug(networking_ice) << _checkInPacketsSinceLastReply << "seconds since last domain list request, squelching traffic";
+        nodeList->setDropOutgoingNodeTraffic(true);
```
Comment (howard-stearns) [inline libraries/networking/src/DomainHandler.cpp]: I don't have an opinion on whether this should be here in DomainHandler, or in NodeList::sendDomainServerCheckIn.

I had arbitrarily assumed the latter, where WARNING_CHECKIN_COUNT is defined. Up to you as to where this stays, but if you keep it here, please remove that commented out code where the warning checkin count is.

File: libraries/networking/src/LimitedNodeList.cpp [hunk 6]
```diff
@@ -409,6 +409,13 @@ qint64 LimitedNodeList::sendUnreliablePacket(const NLPacket& packet, const HifiS
     Q_ASSERT_X(!packet.isReliable(), "LimitedNodeList::sendUnreliablePacket",
                "Trying to send a reliable packet unreliably.");
 
+    if(_dropOutgoingNodeTraffic) {
```
Comment (howard-stearns) [inline libraries/networking/src/LimitedNodeList.cpp]: coding standard

File: libraries/networking/src/LimitedNodeList.cpp (hunk 2 above)
Comment (howard-stearns) [reply libraries/networking/src/LimitedNodeList.cpp]: Maybe for now, keep it simple. If this seems to be working in with the suppression in just the one place, then keep it so. BUT.... might warrant a comment that there are other places that are not suppressed.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19992/HighFidelity-Beta-PR15771-1b31b8c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19992/HighFidelity-Beta-Interface-PR15771-1b31b8c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19992/HighFidelity-Beta-PR15771-1b31b8c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19992/HighFidelity-Quest-Beta-PR15771-1b31b8c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19992/HighFidelity-Beta-PR15771-1b31b8c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19992/HighFidelity-Beta-Interface-PR15771-1b31b8c.exe

**Pull request merged**